openstack/sharedfilesystems/v2/shares/requests.go

```diff
@@ -85,3 +83,41 @@ func GetExportLocations(client *gophercloud.ServiceClient, id string) (r GetExpo
 	_, r.Err = client.Get(getExportLocationsURL(client, id), &r.Body, nil)
 	return
 }
+
+// GrantAccessOptsBuilder allows extensions to add additional parameters to the
+// GrantAccess request.
+type GrantAccessOptsBuilder interface {
+	ToGrantAccessMap() (map[string]interface{}, error)
+}
+
+// GrantAccessOpts contains the options for creation of an GrantAccess request.
+// For more information about these parameters, please, refer to the shared file systems API v2,
+// Share Actions, Grant Access documentation
+type GrantAccessOpts struct {
+	// The access rule type that can be "ip", "cert" or "user".
+	AccessType string `json:"access_type"`
+	// The value that defines the access that can be a valid format of IP, cert or user.
+	AccessTo string `json:"access_to"`
+	// The access level to the share is either "rw" or "ro".
+	AccessLevel string `json:"access_level"`
+}
+
+// ToGrantAccessMap assembles a request body based on the contents of a
+// GrantAccessOpts.
+func (opts GrantAccessOpts) ToGrantAccessMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "allow_access")
+}
+
+// GrantAccessRequest will grant access to a Share based on the values in GrantAccessOpts. To extract
+// the GrantAccess object from the response, call the Extract method on the GrantAccessResult.
+func GrantAccessRequest(client *gophercloud.ServiceClient, id string, opts GrantAccessOptsBuilder) (r GrantAccessResult) {
```

Comment:
Adding "Request" (or "Req", or "Res") is a non-starter. We'd like the functions to read well and be succinct. Users will know that this is a request because it starts with a verb ("Grant", which admittedly is very confusing, since "grant" can also be used as a noun). Let's call this `GrantAccess`.